src/Middleware/ConcurrencyLimiter/src/ConcurrencyLimiterMiddleware.cs

```diff
@@ -51,7 +51,7 @@ public async Task Invoke(HttpContext context)
             // Make sure we only ever call GetResult once on the TryEnterAsync ValueTask b/c it resets.
             bool result;
 
-            if (waitInQueueTask.IsCompleted)
+            if (waitInQueueTask.IsCompletedSuccessfully)
```

Comment:
No, this skips the error condition.

Reply:
I was aware of that, I was thinking that this way if an error occurs it would raise here https://github.com/Kahbazi/AspNetCore/blob/ce97530633ba7d2a068e813b7bc2d3451487f0c2/src/Middleware/ConcurrencyLimiter/src/ConcurrencyLimiterMiddleware.cs#L63 and we have a better exception since await is used instead of Result.